types/jsreport-core/index.d.ts

```diff
@@ -65,50 +84,192 @@ declare namespace JsReport {
+        autoTempCleanup?: boolean;
+        /** specifies if jsreport should discover/search installed extensions in project and use them automatically */
+        discover?: boolean;
+        /**
+         * wheter if jsreport should read list of extensions from a previous generated cache or if it should crawl and try to search extensions again, set it to false when you want to always
+         * force crawling node_modules when searching for extensions while starting jsreport
+         * @default true
+         */
+        useExtensionsLocationCache?: boolean;
+        logger?: {
+            silent?: boolean;
+        };
+        /** global single timeout that controls how much a report generation should wait before it times out */
+        reportTimeout?: number;
+        /**
+         * option that enables passing a custom report timeout per request using req.options.timeout. this enables that the caller of the report generation control the report timeout so enable it
+         * only when you trust the caller
+         * @default false
+         */
+        enableRequestReportTimeout?: boolean;
+        /** @default false */
+        allowLocalFilesAccess?: boolean;
+        encryption?: {
+            /** lenght must be 16 characters */
+            secretKey: string;
+            /** @default true */
+            enabled?: boolean;
+        };
+        templatingEngines?: {
+            /** @default 'dedicated-process' */
+            strategy?: EngineStrategy;
+            numberOfWorkers?: number;
+            forkOptions?: {
+                execArgv?: string | string[];
+            },
+            allowedModules?: string | string[];
+            timeout?: number;
+            templateCache?: {
+                max: number;
+                enabled: boolean;
+            }
         };
-        rootDirectory: string;
-        scripts: {
-            allowedModules: string[];
+        store?: {
+            provider?: ReporterOptionsStoreProvider;
         };
-        tasks: Partial<{
-            allowedModules: string[] | string;
-            strategy: "dedicated-process" | "http-server" | "in-process" | string;
-        }>;
-        tempDirectory: string;
+        blobStorge?: {
+            provider?: ReporterOptionsBlobStorageProvider;
+        };
+        extensions?: any;
+        extensionsList?: string[];
+        /** @default true */
+        migrateEntitySetsToFolders?: boolean;
+        connectionString?: string;
+    }
+
+    interface Settings {
+        add(key: string, value: any, req: Request): any;
+        get(key: string): [SettingItem];
+        findValue(key: string, req: Request): any;
+        set(key: string, avalue: any, req: Request): any;
+        addOrSet(key: string, avalue: any, req: Request): number | null;
+        init(documentStore: any, authorization: any): any;
+        _collection: [SettingItem];
```

Comment:
Is this only ever a single item, or did you mean to write this instead:
```suggestion
        _collection: SettingItem[];
```